src/main/kotlin/wu/seal/jsontokotlin/model/builder/IKotlinDataClassCodeBuilder.kt

```diff
@@ -11,13 +11,15 @@ interface IKotlinDataClassCodeBuilder : ICodeBuilder<DataClass> {
     fun DataClass.genBody(): String
     fun DataClass.genPrimaryConstructorProperties(): String
 
-    fun DataClass.genPrimaryConstructor(): String {
-        return buildString {
-            val primaryConstructorPropertiesCode = genPrimaryConstructorProperties()
-            if (primaryConstructorPropertiesCode.isNotBlank()) {
-                append("(").append("\n")
-                append(primaryConstructorPropertiesCode)
-                append(")")
+    fun genPrimaryConstructor(clazz: DataClass): String {
```

Comment:
Why change to use data class object as input params?

Reply:
> Why change to use data class object as input params?

Let's talk about the former `DataClass.genPrimaryConstructor()`. 
If we use it as extension function, it will be replaced by `kotlinDataClassCodeBuilder` [abbr **Delegator**] (the `delegator` in `BaseDataClassCodeBuilder` [abbr **Base**])  . 
When we use `DataClassCodeBuilderForNoConstructorMemberFields` class, the `genPrimaryConstructorProperties()` method in it will not be invoked. It'is because the **Delegator** in **Base** invoke it `DataClass.genPrimaryConstructor()` method (the **executor** of `genPrimaryConstructor()` method is **Delagator**). And the `genPrimaryConstructorProperties()` method in `DataClass.genPrimaryConstructor()` who's executor is same as `Delegator`.

Let's show an example, 
``` kotlin
//kotlin
interface A {
    fun B.callIn()
    fun B.callOut() {
        callIn()
    }
    fun fun(B b) {
        b.callOut()
    }
}

class C: A {
    fun B.callIn() { print("C callIn") }
}

class D(val delegator: A): A by delegator {
    fun B.callIn() { print("D callIn") }
}
```
``` java 
//java
interface A {
    void callIn(B b);
    void callOut(B b);

    public static final class DefaultImpls {
        public static void callOut(A $this, B b) {
            $this.callIn(b);
        }
        public static void fun(A $this, B b) {
            $this.callback(b);
        }
    }
}
class C implements A {
    public void callIn(B b) { System.out.print("C callIn"); }
    public void callOut(B b) {
        DefaultImpls.callOut(this, b);
    }
    public void fun(B b) {
        DefaultImpls.fun(this, b);
    }
}
class D implements A {
    A delegator;
    public D(A delegator) { this.delegator = delegator; }
    public void callIn(B b) { System.out.print("B callIn"); }
    public void callOut(B b) {
        this.delegator.callOut(b);  //the executor is delegator
}
```